### PR TITLE
before checking pod status, make sure the pod is running

### LINKED ISSUE #107374: [sig-network] Proxy version v1 A set of valid responses are returned for both pod and service ProxyWithPath [Conformance]
##### Error text:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:630
Jan  4 07:05:01.070: Pod didn't start within time out period
Unexpected error:
    <*errors.errorString | 0xc0001ca2b0>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/proxy.go:314
```
#### Recent failures:
[2022/1/6 上午1:25:16 ci-kubernetes-kind-conformance-parallel](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-kind-conformance-parallel/1478778946777518080)
[2022/1/6 上午12:43:32 ci-kubernetes-e2e-windows-containerd-gce-20h2](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-containerd-gce-20h2/1478769130575237120)
[2022/1/5 下午11:03:53 ci-kubernetes-e2e-windows-gce-20h2-1-23](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce-20h2-1-23/1478744049920774144)
[2022/1/5 下午7:14:51 ci-kubernetes-e2e-windows-gce-2019](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce-2019/1478686417042280448)
[2022/1/5 下午3:14:51 ci-kubernetes-e2e-windows-gce-2019](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce-2019/1478626018292928512)

/kind failing-test

/sig network

### PR DESCRIPTION
**Pull request by pacoxu**
/sig network
/kind flake
Fixes #107374

```release-note
None
```

When I check the kubelet log at the failing period, I find a lot of logs like below
```
Jan 05 17:41:17 kind-worker2 kubelet[260]: I0105 17:41:17.039139     260 desired_state_of_world_populator.go:246] "Pod still has one or more containers in the non-exited state and will not be removed from desired state" pod="gc-2728/simpletest.rc-gcdmf"
```
I suppose that this may cause pod creating hanging.  In my env with high load(many pods and namespaces), we tuned the kubelet flag `-kube-api-qps=20 --kube-api-burst=4` to make sure pod creating will not be hung by other apiserver requests. But here, for conformance testing, we may loosen the timeout or add a running pod check before checking pod status.

### CODE REVIEW

**Comment by pacoxu** [comment]
try to run it again （flake in https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/104429/pull-kubernetes-e2e-kind-ipv6/1478993988303196160 just now）
/test pull-kubernetes-e2e-kind-ipv6

File: test/e2e/network/proxy.go [hunk 1]
```diff
@@ -309,9 +310,10 @@ var _ = common.SIGDescribe("Proxy", func() {
 					RestartPolicy: v1.RestartPolicyNever,
 				}}, metav1.CreateOptions{})
 			framework.ExpectNoError(err, "failed to create pod")
-
+			framework.ExpectNoError(e2epod.WaitForPodNameRunningInNamespace(f.ClientSet, "agnhost", f.Namespace.Name))
+			framework.Logf("Pod is running now.")
 			err = wait.PollImmediate(podRetryPeriod, podRetryTimeout, checkPodStatus(f, "test=response"))
-			framework.ExpectNoError(err, "Pod didn't start within time out period")
+			framework.ExpectNoError(err, "Pod did not response with correct status")
```
Comment (aojea) [inline test/e2e/network/proxy.go:316]: we have the same problem in the next test then

https://github.com/kubernetes/kubernetes/pull/107375/files#diff-53d8ea346826e0686770fbee5d0e48c16e21d146b8184cf40ebd9915ab627398R409

File: test/e2e/network/proxy.go [hunk 2]
```diff
@@ -309,9 +310,10 @@ var _ = common.SIGDescribe("Proxy", func() {
 					RestartPolicy: v1.RestartPolicyNever,
 				}}, metav1.CreateOptions{})
 			framework.ExpectNoError(err, "failed to create pod")
-
+			framework.ExpectNoError(e2epod.WaitForPodNameRunningInNamespace(f.ClientSet, "agnhost", f.Namespace.Name))
```
Comment (aojea) [inline test/e2e/network/proxy.go:313]: I wonder if 
```
err = wait.PollImmediate(podRetryPeriod, podRetryTimeout, checkPodStatus(f, "test=response"))
```
is now a duplicate and we should consolidate them in e2epod.WaitForPodNameRunningInNamespac

Comment (aojea) [reply test/e2e/network/proxy.go:313]: WaitForPodNameRunningInNamespace waits 5 min
https://github.com/pacoxu/kubernetes/blob/16adb0f7848d437b86a83b01650f45031f9fd861/test/e2e/framework/pod/wait.go#L52-L53

this has lower timeouts

https://github.com/kubernetes/kubernetes/blob/16adb0f7848d437b86a83b01650f45031f9fd861/test/e2e/network/proxy.go#L61-L62

**Comment by aojea** [comment]
I think we should use the framework functions to wait for the pods running to inherit the e2e timeouts and not use the custom waiting function.
We should also fix the other test with the same structure.

File: test/e2e/network/proxy.go (hunk 2 above)
Comment (pacoxu) [reply test/e2e/network/proxy.go:313]: I change it to check the status after `WaitForPodNameRunningInNamespace`.

File: test/e2e/network/proxy.go (hunk 1 above)
Comment (pacoxu) [reply test/e2e/network/proxy.go]: Update the next test as well.

File: test/e2e/network/proxy.go [hunk 3]
```diff
@@ -487,24 +491,22 @@ func validateRedirectRequest(client *http.Client, redirectVerb string, urlString
 	framework.ExpectEqual(resp.StatusCode, 301, "The resp.StatusCode returned: %d", resp.StatusCode)
 }
 
-func checkPodStatus(f *framework.Framework, label string) func() (bool, error) {
-	return func() (bool, error) {
-		var err error
```
Comment (aojea) [inline test/e2e/network/proxy.go:492]: if we are going to keep it I prefer to keep the same way it was, so an sporadic API call failure doesn't fail the test ... if the first call succeed wait.PollImmediate will have the same effect as this change

Comment (aojea) [reply test/e2e/network/proxy.go:492]: I think that the net change is just adding 
```
			framework.ExpectNoError(e2epod.WaitForPodNameRunningInNamespace(f.ClientSet, "agnhost", f.Namespace.Name))
			framework.Logf("Pod is running now.")
```

to both tests and leaving the rest as is, sorry for the back and forth :smile:

**Pull request closed without merging**